straxen/plugins/event_processing.py

```diff
@@ -222,103 +248,159 @@ def infer_dtype(self):
                 (f'alt_s2_y_{algo}', np.float32,
                  f'Alternate S2 {algo}-reconstructed Y position, uncorrected [cm]')]
 
-        return dtype
+        return posrec_dtpye
+
+    @staticmethod
+    def set_nan_defaults(buffer):
+        """
+        When constructing the dtype, take extra care to set values to
+        np.Nan / -1 (for ints) as 0 might have a meaning
+        """
+        for field in buffer.dtype.names:
+            if np.issubdtype(buffer.dtype[field], np.integer):
+                buffer[field][:] = -1
+            else:
+                buffer[field][:] = np.nan
+
+    def compute(self, events, peaks):
+        result = np.zeros(len(events), dtype=self.dtype)
+        self.set_nan_defaults(result)
+
+        split_peaks = strax.split_by_containment(peaks, events)
+        self.fill_events(result, events, split_peaks)
+        return result
+
+    # If copy_largest_peaks_in_events is ever numbafied, also numbafy this function
+    def fill_events(self, result_buffer, events, split_peaks):
+        """Loop over the events and peaks within that event"""
+        for event_i, _ in enumerate(events):
+            peaks_in_event_i = split_peaks[event_i]
+            n_peaks = len(peaks_in_event_i)
+
+            result_buffer[event_i]['time'] = events[event_i]['time']
+            result_buffer[event_i]['endtime'] = events[event_i]['endtime']
+            result_buffer[event_i]['n_peaks'] = n_peaks
 
-    def compute_loop(self, event, peaks):
-        result = dict(n_peaks=len(peaks),
-                      time=event['time'],
-                      endtime=strax.endtime(event))
-        posrec_save = [(xy + algo)
-                       for xy in ['x_', 'y_']
-                       for algo in self.pos_rec_labels]
+            if not n_peaks:
+                continue
```

Comment:
Should not we raise an error here? An event without peaks should not be possible unless it is a bug is not it?

Reply:
Valid point 👍